ldk/javascript/src/whisper/mapper.ts

```diff
@@ -188,6 +188,18 @@ export function mapToInternalChildComponent(
           component.onChange(error, param, mapToExternalWhisper(whisper, stateMap));
         },
       } as OliveHelps.DateTimeInput;
+    case WhisperComponentType.Icon:
+      // eslint-disable-next-line
```

Comment:
What do you think if we just disable a specific rule for a whole file here?

Reply:
According to the docs for "no-case-declarations", eslint can be satisfied here (with its default config) if we surround cases which declare variables with {} - so Icon could become:

```typescript
case WhisperComponentType.Icon: {
      const onIconClick = component.onClick;
      if (onIconClick) {
        return {
          ...component,
          onClick: (error, whisper) => {
            onIconClick(error, mapToExternalWhisper(whisper, stateMap));
          },
        } as OliveHelps.Icon;
      }
      return component as OliveHelps.Icon;
    }
```

What does the team think of this?

Reply:
Interesting, I have not thought of this. But here is another way to handle it. It doesn't have to be part of your card Cory. Use a map instead of the switch/case. Map<WhisperComponentType, ComponentHandler>. Then trigger handler map.get(component.type).execute(); In this case handler logic will be less coupled and you could abstract it to another file if needed.

Reply:
I got you. Lets work on a refactor for this as part of another card.